text/0000-cargo-features2.md

```diff
@@ -0,0 +1,519 @@
+easy to understand, and ensures that packages are only built once during a
+single build. However, this has problems when features introduce unwanted
+behavior, dependencies, or other requirements. The following three situations
+illustrate some of the unwanted feature unification that the new resolver aims
+to solve:
+
+* Unused targets: If a dependency shows up multiple times in the resolve
+  graph, and one of those situations is a target-specific dependency, the
+  features it enables are enabled on all platforms. See [target
+  dependencies](#target-dependencies) below for how this problem is solved.
+
+* Dev-dependencies: If a dependency is shared as a normal dependency and a
+  dev-dependency, then any features enabled on the dev-dependency will also
+  show up when used as a normal dependency. This only applies to workspace
+  packages; dev-dependencies in packages on registries like [crates.io] have
+  always been ignored. `cargo install` has also always ignored
+  dev-dependencies. See [dev-dependencies](#dev-dependencies) below for how
+  this problem is solved.
+
+* Host-dependencies: Similarly to dev-dependencies, if a build-dependency or
+  proc-macro has a shared dependency with a normal dependency, then the
+  features are unified with the normal dependency. See [host
+  dependencies](#host-dependencies) below for how this problem is solved.
+
+[crates.io]: https://crates.io/
+
+## Command-line feature selection
+
+Cargo has several flags for choosing which features are enabled during a
+build. `--features` allows enabling individual features, `--all-features`
+enables all features, and `--no-default-features` ensures the "default"
+feature is not automatically enabled.
+
+These are fairly straightforward when used with a single package, but in a
+workspace the current behavior is limited and confusing. There are several
+problems in a workspace:
+
+* `cargo build -p other_member --features …` — The listed features are for the
+  package in the current directory, even if that package isn't being built!
+  This also makes it difficult or impossible to build multiple packages at
+  once with different features enabled.
+* `--features` and `--no-default-features` flags are not allowed in the root
+  of a virtual workspace.
+
+See [New command-line behavior](#new-command-line-behavior) below for how
+these problems are solved.
+
+# Guide-level explanation
+
+## New resolver behavior
+
+When the new feature resolver is enabled, features are not always unified when
+a dependency appears multiple times in the dependency graph. The new behaviors
+are described below.
+
+For [target dependencies](#target-dependencies) and
+[dev-dependencies](#dev-dependencies), the general rule is, if a dependency is
+not built, it does not affect feature resolution. For [host
+dependencies](#host-dependencies), the general rule is that packages used for
+building (like proc-macros) do not affect the packages being built.
```

Comment:
Might be a good idea again to explicitly mention that workspace feature unification is deffered to future work, it took me a couple of reads to find out if this is in scope for the RFC.